crates/tower-cmd/src/lib.rs

```diff
@@ -23,8 +27,41 @@ impl App {
         Self { cmd, session }
     }
 
+    async fn check_latest_version() -> Result<Option<String>> {
+        let client = reqwest::Client::new();
+        let resp = client
+            .get("https://pypi.org/pypi/tower-cli/json")
+            .send()
+            .await?;
+            
+        if resp.status().is_success() {
+            let json: Value = resp.json().await?;
+            if let Some(version) = json.get("info").and_then(|info| info.get("version")).and_then(|v| v.as_str()) {
+                return Ok(Some(version.to_string()));
+            }
+        }
```

Comment:
Debug message if this fails might be good. Another argument for pushing all of this to after initialization (see other comment).